packages/expo/src/Notifications/Notifications.ts

```diff
@@ -127,275 +104,224 @@ if (Platform.OS === 'android') {
-        );
+/* Schedule a notification at a later date */
+export async function scheduleLocalNotificationAsync(
+  notification: LocalNotification,
+  options: {
+    time?: Date | number;
+    repeat?: 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
+    intervalMs?: number;
+  } = {}
+): Promise<LocalNotificationId> {
+  // set now at the beginning of the method, to prevent potential weird warnings when we validate
+  // options.time later on
+  const now = Date.now();
+
+  // Validate and process the notification data
+  _validateNotification(notification);
+  let nativeNotification = _processNotification(notification);
+
+  // Validate `options.time`
+  if (options.time) {
+    let timeAsDateObj: Date | null = null;
+    if (options.time && typeof options.time === 'number') {
+      timeAsDateObj = new Date(options.time);
+      if (timeAsDateObj.toString() === 'Invalid Date') {
+        timeAsDateObj = null;
       }
-
-      options = {
-        ...options,
-        time: timeAsDateObj.getTime(),
-      };
+    } else if (options.time && options.time instanceof Date) {
+      timeAsDateObj = options.time;
     }
 
-    if (options.intervalMs != null && options.repeat != null) {
-      throw new Error(`Pass either the "repeat" option or "intervalMs" option, not both`);
+    // If we couldn't convert properly, throw an error
+    if (!timeAsDateObj) {
+      throw new Error(
+        `Provided value for "time" is invalid. Please verify that it's either a number representing Unix Epoch time in milliseconds, or a valid date object.`
+      );
     }
 
-    // Validate options.repeat
-    if (options.repeat != null) {
-      const validOptions = new Set(['minute', 'hour', 'day', 'week', 'month', 'year']);
-      if (!validOptions.has(options.repeat)) {
-        throw new Error(
-          `Pass one of ['minute', 'hour', 'day', 'week', 'month', 'year'] as the value for the "repeat" option`
-        );
-      }
+    // If someone passes in a value that is too small, say, by an order of 1000 (it's common to
+    // accidently pass seconds instead of ms), display a warning.
+    if (timeAsDateObj.getTime() < now) {
+      console.warn(
+        `Provided value for "time" is before the current date. Did you possibly pass number of seconds since Unix Epoch instead of number of milliseconds?`
+      );
     }
 
-    if (options.intervalMs != null) {
-      if (Platform.OS === 'ios') {
-        throw new Error(`The "intervalMs" option is not supported on iOS`);
-      }
+    options = {
+      ...options,
+      time: timeAsDateObj.getTime(),
+    };
+  }
 
-      if (options.intervalMs <= 0 || !Number.isInteger(options.intervalMs)) {
-        throw new Error(
-          `Pass an integer greater than zero as the value for the "intervalMs" option`
-        );
-      }
+  if (options.intervalMs != null && options.repeat != null) {
+    throw new Error(`Pass either the "repeat" option or "intervalMs" option, not both`);
```

Comment:
So if these options are exclusive, let's mark that in method signature.
```typescript
export async function scheduleLocalNotificationAsync(
  notification: LocalNotification,
  options: {
    time?: Date | number;
    repeat?: 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
  } | {
    time?: Date | number;
    intervalMs?: number
  } = {}
): Promise<LocalNotificationId> {
```

Reply:
Unfortunately TypeScript is would not cause error in scenario when someone combines both parts from union type, so we need to keep this runtime check 😞